src/components/Editor/Tab.js

```diff
@@ -0,0 +1,217 @@
+import React, { PureComponent } from "react";
+
+import { showMenu, buildMenu } from "devtools-contextmenu";
+
+import CloseButton from "../shared/Button/Close";
+
+import type { List } from "immutable";
+import type { SourceRecord } from "../../reducers/sources";
+
+import { getFilename, getFileURL, isPretty } from "../../utils/source";
+import { copyToTheClipboard } from "../../utils/clipboard";
+import classnames from "classnames";
+
+type SourcesList = List<SourceRecord>;
+
+type Props = {
+  tabSources: SourcesList,
+  selectSource: Object => void,
+  selectedSource: SourceRecord,
+  closeTab: string => void,
+  closeTabs: (List<string>) => void,
+  togglePrettyPrint: string => void,
+  showSource: string => void,
+  source: SourceRecord,
+  sourceTabsMetaData: {
+    [key: string]: SourceMetaDataMap
+  }
+};
+
+export default class Tab extends PureComponent<Props, State> {
```

Comment:
I like that tabs are now their own component!

Reply:
One thing though. This component takes lots of props that are from Redux but are being passed through in `src/components/Editor/Tabs`, maybe instead of default exporting the class we bring in redux `connect()`.

Reply:
good idea!